pkg/build/api/types.go

```diff
@@ -328,6 +328,9 @@ type GitBuildSource struct {
 
 	// HTTPSProxy is a proxy used to reach the git repository over https
 	HTTPSProxy *string
+
+	// URLCheckTimeoutSeconds is the timeout in seconds to use when checking connectivity to the git server
+	URLCheckTimeoutSeconds *int
```

Comment:
> I don't see a point in this check having a distinct timeout from the build
> timeout.  If you're willing to wait 30 minutes for a build, why aren't you
> willing to wait 30m for git ls remote?

As I recall, this came up recently when a build was doomed to failure due to an invalid host, but an ISP intercepted DNS and gave a bogus address.